### ISSUE TITLE
Cannot authenticate with PUT

### ISSUE THREAD

**Issue by CharlieReitzel** [YvesR/mod_authn_ntlm#18]
Hello Yves,

GET works fine.  The NTLM handshake completes on the same socket connection.

With PUT, however, the socket is closed with each challenge (Connection: close response header is sent).

On Apache, KeepAlive is not specified (default is on) and KeepAliveTimeout is set to 60 seconds (for debugging purposes).

Is this a known issue?  Is it possible to get this to work?

Thanks,
Charlie

**Comment by CharlieReitzel** [comment]
or POST

**Comment by YvesR** [comment]
Never tested this, I will take a look when I find some times next days...

**Comment by CharlieReitzel** [comment]
Ok, another data point: we had a test case that never worked with NTLM, namely a PUT (or POST) with the Expect: 100-Continue request header.   Currently, I am seeing this header sent with _all_ PUT and POST requests using NTLM.   Note: we are seeing a change in behavior between Apache client 4.1.3 and 4.5.2.   I tested a POST using a browser and it works fine, but does _not_ use the Expect: 100-Continue header.   I think this may be the real issue.

**Comment by JBlond** [comment]
> Apache client 4.1.3 and 4.5.2

What kind of client do you use? 4.x.x isn't a valid apache version number.

**Comment by CharlieReitzel** [comment]
Yes, those were valid releases of the Apache HttpComponents releases.
See http://hc.apache.org/downloads.cgi for 4.5.x (currently at 4.5.3 for HttpClient and 4.4.7 for HttpCore).

Earlier versions are still available, e.g.:
https://mvnrepository.com/artifact/org.apache.httpcomponents/httpclient/4.5.2

For 4.1.x, see http://archive.apache.org/dist/httpcomponents/httpclient/

hth
